awsx/ecr/image.ts

```diff
@@ -43,48 +43,43 @@ export function computeImageFromAsset(
   // the unique image name we pushed to.  The name will change if the image changes ensuring
   // the TaskDefinition get's replaced IFF the built image changes.
 
-  const dockerBuild: docker.DockerBuild = {
-    args: dockerInputs.args,
-    cacheFrom: dockerInputs.cacheFrom ? { stages: dockerInputs.cacheFrom } : undefined,
-    context: dockerInputs.path,
-    dockerfile: dockerInputs.dockerfile,
-    env: dockerInputs.env,
-    extraOptions: dockerInputs.extraOptions,
-    target: dockerInputs.target,
-  };
+  const ecrCredentials = aws.ecr.getCredentialsOutput(
+    { registryId: registryId },
+    { parent, async: true },
+  );
+
+  const registryCredentials = ecrCredentials.authorizationToken.apply((authorizationToken) => {
+    const decodedCredentials = Buffer.from(authorizationToken, "base64").toString();
+    const [username, password] = decodedCredentials.split(":");
+    if (!password || !username) {
+      throw new Error("Invalid credentials");
+    }
+    return {
+      registry: ecrCredentials.proxyEndpoint,
+      username: username,
+      password: password,
+    };
+  });
 
-  const uniqueImageName = docker.buildAndPushImage(
+  const dockerImageArgs: docker.ImageArgs = {
     imageName,
-    dockerBuild,
-    repositoryUrl,
-    parent,
-    () => {
-      // Construct Docker registry auth data by getting the short-lived authorizationToken from ECR, and
-      // extracting the username/password pair after base64-decoding the token.
-      //
-      // See: http://docs.aws.amazon.com/cli/latest/reference/ecr/get-authorization-token.html
-      if (!registryId) {
-        throw new Error("Expected registry ID to be defined during push");
-      }
-
-      const credentials = aws.ecr.getCredentialsOutput(
-        { registryId: registryId },
-        { parent, async: true },
-      );
-      return credentials.authorizationToken.apply((authorizationToken) => {
-        const decodedCredentials = Buffer.from(authorizationToken, "base64").toString();
-        const [username, password] = decodedCredentials.split(":");
-        if (!password || !username) {
-          throw new Error("Invalid credentials");
-        }
-        return {
-          registry: credentials.proxyEndpoint,
-          username: username,
-          password: password,
-        };
-      });
+    build: {
+      args: dockerInputs.args,
+      cacheFrom: dockerInputs.cacheFrom
+        ? {
+            images: dockerInputs.cacheFrom,
+          }
+        : undefined,
+      platform: dockerInputs.platform,
+      target: dockerInputs.target,
+      builderVersion: dockerInputs.builderVersion,
     },
-  );
+    registry: registryCredentials,
+  };
+
+  const image = new docker.Image(`image`, dockerImageArgs, { parent });
+
+  const uniqueImageName = image.imageName;
```

Comment:
Pretty sure this introduces a significant regression because `inageName` doesn’t change on rebuilds. So normal examples that use this with ECS will not update the service when the image is re-pushed. 

I believe we have to use `.repoDigest` to get correct behavior.